query-engine/connector-test-kit-rs/query-tests-setup/src/lib.rs

```diff
@@ -292,20 +339,49 @@ fn run_connector_test_impl(
         .unwrap();
         let schema_id = runner.schema_id();
 
-        if let Err(err) = test_fn(runner)
+        let expected_to_fail = ignore_lists::is_expected_to_fail(&full_test_name);
+
+        let test_future = if expected_to_fail {
+            Either::Left(async {
+                match AssertUnwindSafe(test_fn(runner)).catch_unwind().await {
+                    Ok(Ok(_)) => panic!("expected this test to fail but it succeeded"),
```

Comment:
```suggestion
                    Ok(Ok(_)) => panic!("expected this test to fail but it succeeded"),
```
might be worth mentioning the test name here